core/src/main/java/org/axonframework/eventsourcing/eventstore/jdbc/JdbcEventStorageEngine.java

```diff
@@ -413,7 +413,17 @@ protected PreparedStatement readSnapshotData(Connection connection, String ident
      * @throws SQLException when an exception occurs while creating the event data
      */
     protected DomainEventData<?> getDomainEventData(ResultSet resultSet) throws SQLException {
-        return (DomainEventData<?>) getTrackedEventData(resultSet, null);
+        long globalSequence = resultSet.getLong(schema.globalIndexColumn());
+        TrackingToken trackingToken = new GlobalSequenceTrackingToken(globalSequence);
```

Comment:
Won't this generate a ClassCastException when fetching the next batch based on this event (see https://github.com/AxonFramework/AxonFramework/pull/261/files#diff-6df699ec811327f1e444eac5f80f6530R395 line 395).
I think the value for "lowestGlobalSequence" should be set by the user to prevent this error.